micropip/package_index.py

```diff
@@ -157,13 +157,16 @@ def _compatible_wheels(
                 # This key is not available in the Simple API HTML response, so this field may be None
                 size = file.get("size")
 
+                yanked = file.get("yanked")
```

Comment:
I'd say you could just do:
```py
if file.get("yanked"):
    continue
```
here.

Reply:
I guess this wouldn't allow for

> unless they are the only file that matches a version specifier that “pins” to an exact version using either == (without any modifiers that make it a range, such as .*) or ===. 

So maybe handling later is better.

Reply:
Yes I think that rule would be implemented in `find_wheel` so the `yanked` value should be kept in `WheelInfo` and handled there.